dds/DCPS/security/AuthenticationBuiltInImpl.cpp

```diff
@@ -332,7 +332,7 @@ ::DDS::Security::ValidationResult_t AuthenticationBuiltInImpl::begin_handshake_r
     return DDS::Security::VALIDATION_FAILED;
   }
 
-  LocalParticipantData& local_data = *(handshake_data.first);
+  const LocalParticipantData& local_data = *(handshake_data.first);
```

Comment:
```suggestion
  const LocalParticipantData& local_data = *handshake_data.first;
```
If you're going to change the line...